### PR TITLE
DM-27833: Enable gen3 inter-chip crosstalk for DECam

### PR DESCRIPTION
**Pull request by czwa**
Improve inter-chip crosstalk application so it works.

### CODE REVIEW

File: python/lsst/ip/isr/crosstalk.py [hunk 1]
```diff
@@ -717,20 +720,31 @@ def run(self, exposure, crosstalk=None,
 
             if crosstalk.interChip:
                 if crosstalkSources:
+                    # Parse crosstalkSources: Identify which detectors we have available
+                    if isinstance(crosstalkSources[0], lsst.afw.image.Exposure):
+                        # Received afwImage.Exposure
+                        sourceNames = [exp.getDetector().getName() for exp in crosstalkSources]
+                    else:
```
Comment (mrawls) [inline python/lsst/ip/isr/crosstalk.py:727]: To be a little better at catching weird situations, I suggest making this an `elif` and having an `else` which raises.

File: python/lsst/ip/isr/crosstalk.py [hunk 2]
```diff
@@ -717,20 +720,31 @@ def run(self, exposure, crosstalk=None,
...
                         if detName not in sourceNames:
                             self.log.warn("Crosstalk lists %s, not found in sources: %s",
                                           detName, sourceNames)
                             continue
+                        # Get the coefficients.
                         interChipCoeffs = crosstalk.interChip[detName]
+
                         sourceExposure = crosstalkSources[sourceNames.index(detName)]
+                        if not isinstance(sourceExposure, lsst.afw.image.Exposure):
+                            # Dereference the dafButler.DeferredDatasetHandle.
```
Comment (mrawls) [inline python/lsst/ip/isr/crosstalk.py:742]: Suggest explicitly checking to see if it's an instance of `dafButler.DeferredDatasetHandle`, not just checking to see if it's *not* an exposure

File: python/lsst/ip/isr/crosstalk.py [hunk 3]
```diff
@@ -717,20 +720,31 @@ def run(self, exposure, crosstalk=None,
...
+                            sourceExposure = sourceExposure.get()
+
+                        self.log.warn("Correcting detector %s with ctSource %s",
```
Comment (mrawls) [inline python/lsst/ip/isr/crosstalk.py:745]: Is `warn` the best level here? This seems like a good thing to me 😁

**Review by mrawls — COMMENTED** [review]
Dare I ask if there are any crosstalk tests which need augmenting (or writing in the first place)?

File: python/lsst/ip/isr/crosstalk.py (hunk 2 above)
Comment (czwa) [reply python/lsst/ip/isr/crosstalk.py:742]: I've done this, and changed the `isinstance(sourceExposure, lsst.afw.image.Exposure)` block to raise as well.

File: python/lsst/ip/isr/crosstalk.py (hunk 3 above)
Comment (czwa) [reply python/lsst/ip/isr/crosstalk.py:745]: Updated to `info`, because that should be less confusing.

**Comment by czwa** [comment]
> Dare I ask if there are any crosstalk tests which need augmenting (or writing in the first place)?

I've added a test case where we only have inter-chip crosstalk from an exposure passed in.

**Review by mrawls — APPROVED** [review]

**Pull request merged**